### PR TITLE
Mount livecd in /run/initramfs/live under uki

### PR DESCRIPTION
**Pull request by Itxaka**
This also brings a small tweak to the log-to-file logger to skip the color so the logging in the files dont look like crap

### CODE REVIEW

**Comment by Itxaka** [comment]
Tested and working

File: pkg/mount/dag_steps.go [hunk 1]
```diff
@@ -715,3 +715,42 @@ func (s *State) UKIUnlock(g *herd.Graph, opts ...herd.OpOption) error {
 		return kcrypt.UnlockAll(true)
 	}))...)
 }
+
+// MountLiveCd tries to mount the livecd if we are booting from one into /run/initramfs/live
+// to mimic the same behavior as the livecd on non-uki boot.
+func (s *State) MountLiveCd(g *herd.Graph, opts ...herd.OpOption) error {
+	return g.Add(cnst.OpUkiMountLivecd, append(opts, herd.WithCallback(func(ctx context.Context) error {
+		// If we are booting from Install Media
+		if !internalUtils.EfiBootFromInstall() {
```
Comment (mauromorales) [inline pkg/mount/dag_steps.go:724]: minor request but I personally would prefer an early return, so the rest of the code is not nested even deeper

File: pkg/mount/dag_steps.go [hunk 2]
```diff
@@ -715,3 +715,42 @@ func (s *State) UKIUnlock(g *herd.Graph, opts ...herd.OpOption) error {
...
+			err := os.MkdirAll(s.path("/run/initramfs/live"), 0755)
+			if err != nil {
+				internalUtils.Log.Err(err).Msg("Creating /run/initramfs/live")
+				return nil
+			}
+			// Try to find the CDROM device by label /dev/disk/by-label/UKI_ISO_INSTALL
+			_, err = os.Stat("/dev/disk/by-label/UKI_ISO_INSTALL")
```
Comment (mauromorales) [inline pkg/mount/dag_steps.go:731]: maybe this string can be moved to a constant

**Comment by mauromorales** [comment]
is this one related to kairos-io/kairos#2224?

**Comment by Itxaka** [comment]
> is this one related to [kairos-io/kairos#2224](https://github.com/kairos-io/kairos/issues/2224)?

unfortunately not, but it migth fix it as a result?

File: pkg/mount/dag_steps.go (hunk 2 above)
Comment (Itxaka) [reply pkg/mount/dag_steps.go:731]: yep, will do

File: pkg/mount/dag_steps.go (hunk 1 above)
Comment (Itxaka) [reply pkg/mount/dag_steps.go]: This is what you mentioned or did I got this totally incorrect?

File: pkg/mount/dag_steps.go [hunk 3]
```diff
@@ -715,3 +715,42 @@ func (s *State) UKIUnlock(g *herd.Graph, opts ...herd.OpOption) error {
...
+
+// MountLiveCd tries to mount the livecd if we are booting from one into /run/initramfs/live
+// to mimic the same behavior as the livecd on non-uki boot.
+func (s *State) MountLiveCd(g *herd.Graph, opts ...herd.OpOption) error {
+	return g.Add(cnst.OpUkiMountLivecd, append(opts, herd.WithCallback(func(ctx context.Context) error {
+		// If we are booting from Install Media
+		if internalUtils.EfiBootFromInstall() {
+			internalUtils.Log.Debug().Msg("Not mounting livecd as we think we are booting from removable media")
+		} else {
+			err := os.MkdirAll(s.path(cnst.UkiLivecdMountPoint), 0755)
```
Comment (mauromorales) [inline pkg/mount/dag_steps.go:727]: I mean more like:

```
if internalUtils.EfiBootFromInstall() {
			internalUtils.Log.Debug().Msg("Not mounting livecd as we think we are booting from removable media")
                        return/break/continue/whatever
		}
		
err := os.MkdirAll(s.path(cnst.UkiLivecdMountPoint), 0755)
```

**Review by mauromorales — APPROVED** [review]
lgtm

**Pull request merged**